docs/source/daml/intro/2_Scenario.rst

```diff
@@ -0,0 +1,146 @@
+
+The first columns, labelled vertically show which parties know about which contracts. In this simple scenario, the sole party "Alice" knows about the contract she created.
+
+The second column shows the ID of the contract. This will be explained later.
+
+The third column shows the status of the contract, either ``active`` or ``archived``. The remaining columns show the contract arguments, with one column per field. As expected, field ``owner`` is ``'Alice'``. The single quotation marks indicate that ``Alice`` is a party.
+
+To run the same test from the command line, save your module in a file ``Intro_2_Scenario.daml`` and run ``daml damlc -- test Intro_2_Scenario.daml``. If your file contains more than one scenario, all of them will be run.
+
+.. _intro_2_failure:
+
+Testing for Failure
+-------------------
+
+In :doc:`1_Token` you learned that creating a ``Token`` requires the authority of its owner. In other words, it should not be possible for Alice to create a Token for another party and vice versa. A reasonable attempt to test that would be:
+
+.. literalinclude:: daml/Intro_2_Scenario.daml
+  :language: daml
+  :start-after: -- FAILURE_TEST_1_BEGIN
+  :end-before: -- FAILURE_TEST_1_END
+
+However, if you open the scenario view for that scenario you, are faced with the following, message:
+
+.. figure:: images/2_Scenario/failure.png
+
+The scenario failed, as expected, but it aborted at the first failure so it only tested that Alice can't create a token for Bob. The second ``submit`` statement was never reached. To test for failing submits and keep the scenario running thereafter, or fail if the submission succeeds, DAML has the ``submitMustFail`` function:
+
+.. literalinclude:: daml/Intro_2_Scenario.daml
+  :language: daml
+  :start-after: -- TOKEN_TEST_2_BEGIN
+  :end-before: -- TOKEN_TEST_2_END
+
+``submitMustFail`` never has an impact on the ledger so the resulting tabular scenario view just shows the two Tokens resulting from the successful ``submit`` statements. Note the new column for Bob as well as the visibilities. Alice and Bob cannot see each others' Tokens.
+
+.. _archiving:
+
+Archiving Contracts
+-------------------
+
+Archival of contracts works just like creation, but using ``archive`` instead of ``create``. Whilst ``create`` takes an instance of a template, ``archive`` takes a reference to a contract.
+
+Contract references have type ``ContractId a`` in DAML, where ``a`` is a *type parameter* representing the type of contract that the id refers to. A reference to a ``Token`` would be a ``ContractId Token``, for example.
+
+To ``archive`` the Token Alice has created, you need to get a handle on its contract id. In scenarios, this is again done using ``<-`` notation, because the contract ID needs to be retrieved from the ledger. How this works is discussed in :doc:`5_Restrictions`. The scenario first checks that Bob cannot archive Alice's Token and then Alice successfully archives it.
```

Comment:
```suggestion
To ``archive`` the Token Alice has created, you need to get a handle on its contract ID. In scenarios, you do this using ``<-`` notation. That's because the contract ID needs to be retrieved from the ledger. How this works is discussed in :doc:`5_Restrictions`. 

This scenario first checks that Bob cannot archive Alice's Token and then Alice successfully archives it:
```